samples/snippets/sample_templates.py

```diff
@@ -0,0 +1,195 @@
+        template_name: name of the new template that is created.
+
+    Returns:
+        InstanceTemplate object describing the new instance template.
+    """
+    # Describe the size and source image of the boot disk to attach to the instance.
+    disk = compute_v1.AttachedDisk()
+    initialize_params = compute_v1.AttachedDiskInitializeParams()
+    initialize_params.source_image = (
+        "projects/debian-cloud/global/images/family/debian-11"
+    )
+    initialize_params.disk_size_gb = 250
+    disk.initialize_params = initialize_params
+    disk.auto_delete = True
+    disk.boot = True
+
+    # Use the default network, without specifying a subnetwork.
+    network_interface = compute_v1.NetworkInterface()
+    network_interface.name = "global/networks/default"
+
+    # Let the instance have external IP
+    network_interface.access_configs = [compute_v1.AccessConfig()]
+    network_interface.access_configs[0].name = "External NAT"
+    network_interface.access_configs[0].type_ = compute_v1.AccessConfig.Type.ONE_TO_ONE_NAT
+    network_interface.access_configs[0].network_tier = (
+        compute_v1.AccessConfig.NetworkTier.PREMIUM
+    )
+
+    template = compute_v1.InstanceTemplate()
+    template.name = template_name
+    template.properties.disks = [disk]
+    template.properties.machine_type = "e2-standard-4"
+    template.properties.network_interfaces = [network_interface]
+
+    template_client = compute_v1.InstanceTemplatesClient()
+    operation_client = compute_v1.GlobalOperationsClient()
+    op = template_client.insert(project=project_id, instance_template_resource=template)
+    operation_client.wait(project=project_id, operation=op.name)
+
+    return template_client.get(project=project_id, instance_template=template_name)
+
+
+def create_template_from_instance(project_id: str, instance: str, template_name: str) -> compute_v1.InstanceTemplate:
+    """
+    Create a new instance template based on an existing instance.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you use.
+        instance: the instance the new template will be based on. This value uses
+            the following format: "projects/{project}/zones/{zone}/instances/{instance_name}"
+        template_name: name of the new template that is created.
+
+    Returns:
+        InstanceTemplate object describing the new instance template.
+    """
+    disk = compute_v1.DiskInstantiationConfig()
+    # This name must match the name of a disk attached to the instance you are
+    # basing your template on.
+    disk.device_name = "disk-1"
+    # Replace the original disk image used in your instance with a Rocky Linux image.
+    disk.instantiate_from = (
+        compute_v1.DiskInstantiationConfig.InstantiateFrom.CUSTOM_IMAGE
+    )
+    disk.custom_image = "projects/rocky-linux-cloud/global/images/family/rocky-linux-8"
+    # You can override the auto_delete setting.
```

Comment:
As written, this raises some questions like "What else can I override?". If this change is specific to this example, we can probably change to just say "Override the auto_delete setting."

Reply:
You can't really override anything else. See [here](https://googleapis.dev/python/compute/latest/compute_v1/types.html#google.cloud.compute_v1.types.DiskInstantiationConfig) to see what the `DiskInstantiationConfig` can fit.

I'll change to comment as suggested.